### PR TITLE
ECF-TECH-36: GitHub actions automated testing

### PR DESCRIPTION
**Pull request by ethanmills**
### Context
We want automated tests to run on every push

### Changes proposed in this pull request
- Add a new workflow to run automated tests
- Fix said tests

### Guidance to review
None really

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,9 @@
     "jest": "^26.6.3",
     "webpack-dev-server": "^3.11.0"
   },
+  "scripts": {
+    "test": "jest --passWithNoTests"
```
Comment (ethanmills) [inline package.json:22]: we'll remove this --passWithNoTests when we have some tests...

File: db/schema.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@
 ActiveRecord::Schema.define(version: 2020_11_30_093110) do
 
   # These are extensions that must be enabled in order to support this database
+  enable_extension "pgcrypto"
```
Comment (kicferk1) [inline db/schema.rb:16]: What does this do?

Comment (ethanmills) [reply db/schema.rb:16]: Required for UUID generation

Comment (kicferk1) [reply db/schema.rb:16]: Ooh, what? That's weird - I thought we added uuids (and tests that should make use of objects using uuids as ids) a while ago :/

Or is this enabling uuids at a database level when our old code does it on application level?

Basically, I'm wondering how did you get an error that needed this fix?

Comment (amazimbe) [reply db/schema.rb:16]: Could be dependent on the db version

Comment (ethanmills) [reply db/schema.rb:16]: This enables the feature in the db that your code depended on

Comment (ethanmills) [reply db/schema.rb:16]: It's off by default on my local db and the db provided by the github action

**Review by kicferk1 — APPROVED** [review]

**Review by amazimbe — APPROVED** [review]

**Pull request merged**